modules/kargoBidAdapter.js

```diff
@@ -28,57 +28,77 @@ export const spec = {
   },
   buildRequests: function(validBidRequests, bidderRequest) {
     const currencyObj = config.getConfig('currency');
-    const currency = (currencyObj && currencyObj.adServerCurrency) || 'USD';
-    const bidIDs = {};
-    const bidSizes = {};
+    const currency = (currencyObj && currencyObj.adServerCurrency);
+    const impressions = [];
 
     _each(validBidRequests, bid => {
-      bidIDs[bid.bidId] = bid.params.placementId;
-      bidSizes[bid.bidId] = bid.sizes;
+      impressions.push(spec.getImpression(bid)) 
     });
 
     const firstBidRequest = validBidRequests[0];
 
     const tdid = deepAccess(firstBidRequest, 'userId.tdid')
 
-    const transformedParams = Object.assign({}, {
-      sessionId: spec._getSessionId(),
-      requestCount: spec._getRequestCount(),
-      timeout: bidderRequest.timeout,
-      currency,
-      cpmGranularity: 1,
-      timestamp: (new Date()).getTime(),
-      cpmRange: {
-        floor: 0,
-        ceil: 20
-      },
-      bidIDs,
-      bidSizes,
+    // Pull Social Canvas segments and embed URL
+    const socialCanvas = deepAccess(firstBidRequest, 'params.socialCanvas');
+ 
+    const krakenParams = Object.assign({}, {
+      pbv: '$prebid.version$',
+      aid: firstBidRequest.auctionId,
+      sid: spec._getSessionId(),
+      url: spec._getAllMetadata(bidderRequest).pageURL,
+      to: bidderRequest.timeout,
+      ts: new Date().getTime(),
       device: {
-        width: window.screen.width,
-        height: window.screen.height
+        size:  { 
+          width: window.screen.width,
+          height: window.screen.height
+        }
```

Comment:
@jsadwith Any objection to making the size object like this rather than the array defined here https://kargo1.atlassian.net/wiki/spaces/KRAK/pages/3171876871/Prebid.js+Adapter+-+Request+Format+Proposal#:~:text=%22size%22%3A%20%5B2560%2C%201440%5D%2C

Reply:
@njflynn ideally, it would be `size:  [ window.screen.width, window.screen.height ]`, but do you see technical limitations to doing so?

Reply:
No technical limitation. The thought occurred to me while considering readability. Will make it the preferred array implementation.